Tips the Scales

Aurorablade opened the pull request:
...Of Unathi.

Okay i am now takeing the suggestions of from the forum (and from talking to dumbdumb) and take some bits.
EVERYTHING IS SUBJECT TO CHANGE..AND MY CODE MAY SUCK WE CAN FIX THAT
:cl: Fethas
tweak:Unathi,Now have a tail attack as a verb, Take a small amount more brute.
tweak: Vulpkanin, Hunger drain increased slightly.
rscadd:Diona now have hair by skittles.
/:cl:


Diona 'hair' sprites by skittles:
![](https://puu.sh/z4LZ9/575ace22bd.gif)
![](https://puu.sh/z4M12/0132a972d1.png)

Tayyyyyyy commented:
Maybe make it so that the tail slap can be used when cuffed, but maybe not when buckled? Also correct me if I'm wrong, but it seems like there's no cooldown on the ability?

Tayyyyyyy commented:
Hmm haven't looked at much cooldown stuff but off the top of my head, maybe have a var on the mob that has the last time you successfully tail slashed, and then subtract the current world.time from it, and if it's less than a certain amount of time, say "wait [world.time - src.last_tail_slash_time] seconds" before you can tail slash again.

Without a cooldown it would be pretty funny to see unathi tail slapping everyone into submission 😄

Oh and maybe it's worth considering a bit of a telegraph before it happens, like "the unathi rears back its tail!" and give em like a second to get out of the way...just throwing out some ideas.

Aurorablade commented:
not sure if i can do that with how our current species code is set up. its kind global. it also why i added a stamina thing to it..it takes samina to use.

Dinarzad commented:
Diona needing water is neat. Makes sense I suppose. Also kinda like the plant hairstyles
Anyway some of their slow down could be toned back slighty (Not to normal move speed just not AS slow) since the regen is being made a bit more conditional then just "Stand in Light, acquire HP" ?

davidchan13 commented:
Does tail slap get a HUD icon?

And as a flavor request, give it a spin animation like when Ian chases his tail (Albeit only a half spin or one spin.)

pinatacolada commented:
id be funny if diona could stand on soil patches and hydroponic trays and absorb their water for a quicker HP regen or feed or some sort of bonus

Birdtalon commented:
I don't think diona need another chemical based vulnerability to be honest. Them needing water to benefit from the healing is decent but they are already super vulnerable to plant-b-gone so I don't think they need a vulnerability to drying agent also.

Aurorablade commented:
i like the slow as is.
The tail icon does not have a hud thing, our racial abilites would need a bit of tweaking to use action buttons. i had the idea about spin also.
On the drying agent, i could just making a nutrition loss instead of also adding the tox bit.

davidchan13 commented:
IPCs have a HUD button for their arm charger though I suppose that counts as an implant.

Removing the tox from drying reagent would be good, as it was specifically suggested as a non-lethal alternative to atrazine.

taukausanake commented:
@davidchan13 What about Skrell headpockets? That's an organ and has an action button. Then again that might only work if tails were counted as an organ _(hint hint)_

MarcellusPye commented:
@Aurorablade The nutrition gained from water is capped at  NUTRITION_LEVEL_WELL_FED+10, or 460. Healing only begins when nutrition >= NUTRITION_LEVEL_FULL, or nutrition >= 550. As nutrition is capped at 460, it's impossible to get to the point where you're full enough to start healing. Unless there's some other source of nutrition you're supposed to use which I can't think of, or I made a mistake, both of which are very possible.

KasparoVy commented:
This sounds and looks cool as hell.  Will definitely look into it further when I'm not on mobile...

KasparoVy commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -78,6 +79,24 @@
+		/mob/living/carbon/human/proc/tail_lash
+		)
+
+/mob/living/carbon/human/proc/tail_lash()
```
Does this have a cooldown? I wonder if you can macro it and lash someone machine-gun style.  Also- perhaps a sound effect would be well applied here?

Aurorablade replied:
i put in a stamina cost. so it drains 15 staimina on use. i forgot the sound effect.

KasparoVy replied:
Still wack being able to slap people as fast as you can get the attacks off even if it'll stun you after a few strikes. Any cooldown is better than none

Aurorablade commented:
i missed the non lethal part. also the charger thing is due to that being an organ. action abilitys on organs are set up to be action buttons...racials are not. at the moment i cam gonna look at it!

KasparoVy commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -78,6 +79,24 @@
+	for(var/mob/living/carbon/human/C in orange(1))
+		var/obj/item/organ/external/E = C.get_organ(pick("l_leg", "r_leg", "l_foot", "r_foot", "groin"))
+		if(E)
+			visible_message("<span class='danger'> [src] smacks [C] in the [E] with thier tail!</span>", "<span class='danger'>You hit [C] with your tail!</span>")
```
Might even be able to add a little animation here to go along with the sound effect where the player 'spins' (not flips) to indicate the lashing (should they not be restrained).

Aurorablade commented:
Removed tox loss on drying agent.
Found out on adjusting nutrituion levels you should do += on some things or it gets odd. I may need to do a further check on it. Will need to be slapped on better logic.
Added some checks to the tail lash and a stamina check, and a sound...and a spin...

Jountax commented:
Did diona really need a nerf? I hardly ever see them in game as it is.

MarcellusPye commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -844,15 +901,15 @@
-
+	if((light_amount >= 5 && H.nutrition >= NUTRITION_LEVEL_FED ) && !H.suiciding) //if there's enough light, heal
+		if(H.nutrition >= NUTRITION_LEVEL_WELL_FED)
+			H.nutrition += min(H.nutrition+light_amount, NUTRITION_LEVEL_FED)
```
This will exponentially increase  nutrition once it's above a certain point, giving ridiculous healing and probably breaking lots of stuff.

MarcellusPye replied:
Sorry, small mistake, it'll just add 350 per tick, not increase exponentially.

Aurorablade replied:
well the healing is meant to be like that..the nutrition bit is what i have to try to balance out.

Aurorablade replied:
open to a suggetion there

EldritchSigma commented:
Thumbs up, this is more like it. Really good PR Auroblade

Aurorablade commented:
on the metabolism thing, i did give vulps a slightly higher hunger drain. I need to look at the other metabolism stuff..but we could add that via a trait system later.
Diona (if i did it right) drain hunger slower.

Humans are kinda ment to be bland, if i get a trait system in we could tweak that out later.

Skrells DO have some things going for them already they are not as bland as you think its just not as obvious as other.

The drying agent only drains nutrition now (so does darkness) reasoning for the drying agent is its sucking the water out of diona.

What we could use though is someone (And i am willing to do it but its gonna be a TASK) to make the species datum non global, as it stands now if you edit the species datum it will effect EVERYONE, we need it something like tg has so instead of .species its .dna.species, in essence its hooked to the players dna. kinda like a holder for the species datum data.

@KasparoVy question. I need to get it so diona hair is uncolored on the character preview, what do? i mean its colored in the preview just not when you get in game.

KasparoVy commented:
The hair being coloured in character preview is a bug it seems. There needs to be checks for the hair datum's `do_colouration` var [here](https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/mob/new_player/preferences_setup.dm#L349), [here](https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/mob/new_player/preferences_setup.dm#L355), [here](https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/mob/new_player/preferences_setup.dm#L373) and [here](https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/mob/new_player/preferences_setup.dm#L378)

tigercat2000 commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -78,6 +79,39 @@
+				to_chat(user, "Rest before tail lashing again!")
+			else
+				user.changeNext_move(CLICK_CD_MELEE)
+				user.visible_message("<span class='danger'> [src] smacks [C] in the [E] with thier tail!</span>", "<span class='danger'>You hit [C] in the [E] with your tail!</span>")
```
No spaces after spans

tigercat2000 replied:
No `the [E]` either

tigercat2000 commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -78,6 +79,39 @@
+		var/obj/item/organ/external/E = C.get_organ(pick("l_leg", "r_leg", "l_foot", "r_foot", "groin"))
+		if(E)
+			if(user.getStaminaLoss() >= 50)
+				to_chat(user, "Rest before tail lashing again!")
```
This should probably be outside of the for loop, or contain a `break` statement so that it isn't spammed for every mob within range

tigercat2000 commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -78,6 +79,39 @@
+				user.visible_message("<span class='danger'> [src] smacks [C] in the [E] with thier tail!</span>", "<span class='danger'>You hit [C] in the [E] with your tail!</span>")
+				user.adjustStaminaLoss(15)
+				C.apply_damage(5, BRUTE, E)
+				user.spin(20,1)
```
Spaces after commas

tigercat2000 commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -78,6 +79,39 @@
+				user.adjustStaminaLoss(15)
+				C.apply_damage(5, BRUTE, E)
+				user.spin(20,1)
+				playsound(user.loc, 'sound/weapons/slash.ogg', 50, 0)
```
This makes me think this should be a verb on carbons, and given to xenomorphs as well (they have sharp tails, don't they?)

KasparoVy replied:
Good point

tigercat2000 commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -838,6 +891,10 @@
 
+/datum/species/diona/water_act(var/mob/living/carbon/C, volume, temperature, source)
+	..()
+	C.nutrition = min(C.nutrition+volume, NUTRITION_LEVEL_WELL_FED+10)
```
Spacing around math operators

tigercat2000 commented on code/modules/reagents/chemistry/reagents/water.dm:
```diff
@@ -29,6 +29,9 @@
 			var/mob/living/carbon/human/H = M
 
+			if(IS_PLANT in H.species.species_traits)
+				H.nutrition = min(H.nutrition+volume, NUTRITION_LEVEL_WELL_FED+10)
```
Spacing around math operators

tigercat2000 replied:
SPACING AROUND MATH OPERATORS
`nutrition + volume`, `FED + 10`

tigercat2000 commented on code/modules/reagents/chemistry/reagents/water.dm:
```diff
@@ -58,6 +61,9 @@
 			var/mob/living/carbon/human/H = M
 
+			if(IS_PLANT in H.species.species_traits)
+				H.nutrition = min(H.nutrition+volume, NUTRITION_LEVEL_WELL_FED+10)
```
Spacing around math operators

tigercat2000 replied:
:eyes: do I need to say it

KasparoVy commented:
Only missed a couple critiques, looking really good

tigercat2000 commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -97,18 +97,18 @@
 		to_chat(user, "You need freedom of movment to tail lash!")
 		return
+	if(user.getStaminaLoss() >= 50)
+		to_chat(user, "Rest before tail lashing again!")
```
Add span classes

tigercat2000 commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -97,18 +97,18 @@
-				user.spin(20,1)
-				playsound(user.loc, 'sound/weapons/slash.ogg', 50, 0)
+			user.changeNext_move(CLICK_CD_MELEE)
+			user.visible_message("<span class='danger'> [src] smacks [C] in [E] with thier tail! </span>", "<span class='danger'>You hit [C] in [E] with your tail! </span>")
```
No starting space in visible messages, please.

tigercat2000 replied:
Or ending space, for that matter.
Also `thier` is spelled `their`.

KasparoVy commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -78,6 +79,39 @@
+/datum/action/innate/tail_lash/Activate()
+	var/mob/living/carbon/human/user = owner
+	if(!user.restrained() || !user.buckled)
+		to_chat(user, "<span class='warning'>You need freedom of movment to tail lash!</span>")
```
movement*

tigercat2000 replied:
^

tigercat2000 commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -78,6 +79,39 @@
+
+
+/datum/species/unathi/handle_post_spawn(var/mob/living/carbon/human/H)
+
```
No newline

tigercat2000 commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -78,6 +79,39 @@
+		var/obj/item/organ/external/E = C.get_organ(pick("l_leg", "r_leg", "l_foot", "r_foot", "groin"))
+		if(E)
+			user.changeNext_move(CLICK_CD_MELEE)
+			user.visible_message("<span class='danger'> [src] smacks [C] in [E] with their tail! </span>", "<span class='danger'>You hit [C] in [E] with your tail!</span>")
```
No space between `danger'> [src]` or `tail! </span>`

tigercat2000 commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -838,27 +891,30 @@
 
-	if((light_amount >= 5) && !H.suiciding) //if there's enough light, heal
-
+	if((light_amount >= 5 && H.nutrition >= NUTRITION_LEVEL_FED ) && !H.suiciding) //if there's enough light, heal
```
No space after `FED )`

tigercat2000 commented on code/modules/reagents/chemistry/reagents/water.dm:
```diff
@@ -514,3 +520,9 @@
 		qdel(O)
 		new /obj/item/clothing/shoes/galoshes/dry(t_loc)
+
+/datum/reagent/drying_agent/reaction_mob(mob/living/M, method=TOUCH, volume)
```
`method = TOUCH`

tigercat2000 commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -838,27 +891,30 @@
-	H.radiation = max(H.radiation-rads, 0)
-	H.nutrition = min(H.nutrition+rads, NUTRITION_LEVEL_WELL_FED+10)
+	H.radiation = max(H.radiation - rads, 0)
+	H.nutrition = min(H.nutrition + rads, NUTRITION_LEVEL_WELL_FED+10)
```
`FED + 10`
